src/amqp10_client_session.erl

```diff
@@ -653,6 +661,42 @@ translate_terminus_durability(none) -> 0;
 translate_terminus_durability(configuration) -> 1;
 translate_terminus_durability(unsettled_state) -> 2.
 
+translate_filters(Filters) when is_map(Filters) andalso map_size(Filters) =< 0 -> undefined;
+translate_filters(Filters) when is_map(Filters) -> {
+    map,
+    maps:fold(
+        fun(<<"apache.org:legacy-amqp-direct-binding:string">> = K, V, Acc) when is_binary(V) ->
+            [{{symbol, K}, {described, {symbol, K}, {utf8, V}}} | Acc];
+        (<<"apache.org:legacy-amqp-topic-binding:string">> = K, V, Acc) when is_binary(V) ->
+            [{{symbol, K}, {described, {symbol, K}, {utf8, V}}} | Acc];
+        (<<"apache.org:legacy-amqp-headers-binding:map">> = K, V, Acc) when is_map(V) ->
+            [{{symbol, K}, {described, {symbol, K}, translate_legacy_amqp_headers_binding(V)}} | Acc];
+        (<<"apache.org:no-local-filter:list">> = K, V, Acc) when is_list(V) ->
+            [{{symbol, K}, {described, {symbol, K}, lists:map(fun(Id) -> {utf8, Id} end, V)}} | Acc];
+        (<<"apache.org:selector-filter:string">> = K, V, Acc) when is_binary(V) ->
```

Comment:
Is crashing when no key matches desirable behavior? I guess on one hand, this will happen at connection time so it's not the worst thing (and we prevent silly typos like `filtre`), but it did surprise me to see.

Reply:
Oh, I guess this is the same behavior as upstream. Disregard!